.github/workflows/ci.yml

```diff
@@ -28,30 +28,24 @@ jobs:
     needs: check_if_skip_duplicate_job
     if: needs.check_if_skip_duplicate_job.outputs.should_skip != 'true'
     steps:
-      #- uses: cachix/install-nix-action@v22
       - uses: actions/checkout@v4
       - run: nix build -L .#checks.x86_64-linux.aeneas-check-tidiness
       - run: nix build -L .#aeneas
       - run: nix build -L .#checks.x86_64-linux.aeneas-tests
       - run: nix build -L .#checks.x86_64-linux.aeneas-verify-fstar
       - run: nix build -L .#checks.x86_64-linux.aeneas-verify-coq
       - run: nix build -L .#checks.x86_64-linux.aeneas-verify-hol4
-      # Lean doesn't work with Nix
+      # Lean cannot run its tests in the sandbox because `elan` will download things
       #- run: nix build -L .#checks.x86_64-linux.aeneas-verify-lean
 
-  lean: # Lean isn't supported by Nix, so we put it in a different job
+  lean:
     runs-on: [ubuntu-latest]
     needs: check_if_skip_duplicate_job
     if: needs.check_if_skip_duplicate_job.outputs.should_skip != 'true'
     steps:
-      # Install curl
-      - run: sudo apt update && sudo apt install curl
-      # Install Elan (https://leanprover-community.github.io/install/linux.html) and Lean in
-      # non-interactive mode:
-      - run: curl https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh -sSf | bash -s -- -y
-      # Checkout the repo and download it to the runner
-      - name: Checkout
-        uses: actions/checkout@v4
-      # Verify - note that we need to update the environment with `source` so
-      # that the lake binary is in the path.
-      - run: source ~/.profile && cd tests/lean && make
+      - uses: actions/checkout@v4
+      - uses: cachix/install-nix-action@v26
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
```

Comment:
Is there an easy way that you can reuse the nixpkgs we use for the other tests?

Reply:
What I could do is simply have a devShells with `elan` from our flake and go from there.

Reply:
(But I'm mostly interested into validating if this idea can work so, using our latest nixpkgs should not matter too much.)

Reply:
Apparently you need dun too, so devshell seems like the way to go